cmake/FindOrBuildOpus.cmake

```diff
@@ -0,0 +1,47 @@
+message(STATUS "Testing opus at ${OPUS_DIR} for FARGAN")
+
+if(OPUS_DIR)
+    # This ensures that the version of opus found on the user's system
+    # has FARGAN enabled (--enable-dred when building opus). If our test
+    # file can't be built, we can't use this version of opus.
+    try_compile(OPUS_HAS_FARGAN
+        SOURCES ${CMAKE_SOURCE_DIR}/cmake/test_opus.c
+        CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${OPUS_DIR}/dnn;${OPUS_DIR}/celt;${OPUS_DIR}/include"
+        LINK_LIBRARIES ${OPUS_DIR}/.libs/libopus.a)
+endif(OPUS_DIR)
```

Comment:
This might be over complicating things - it really is 100% just to build opus here, that will get us a known configuration.

Reply:
Should we have a work item to implement using an existing build later on? That will make it easier for package maintainers as package build environments typically don't have internet access and thus can't checkout other repos during the build process. (For example, the Debian version of freedv-gui currently doesn't have FreeDV Reporter enabled precisely because there's no sioclient-cpp package and Debian's build system isn't able to pull the source code for it from GitHub.)

Reply:
Pls see the scope statement the top of the README - this is experimental code not intended for packaging.

Reply:
OK, updated accordingly.